### ISSUE TITLE
SQ 5.3 and LOC metric

### ISSUE THREAD

**Issue by Bertk** [SonarOpenCommunity/sonar-cxx#784]
The LOC metric does not reflect the lines of code but the lines in the source file.
How can I get the aggregated lines for a function in batch processing?

I was working on the 2 checks from ametselaar which were not merged last year (#397)
I prepared a PR but the no of lines is not handled correctly (see 517097e91533e20bed6f7e41eaf0404f4e1f824b)

**Comment by guwirth** [comment]
Maybe has something to do with #747?

**Comment by guwirth** [comment]
* JavaScript are counting lines between curly braces https://github.com/SonarSource/sonar-javascript/blob/master/javascript-checks/src/main/java/org/sonar/javascript/checks/TooManyLinesInFunctionCheck.java
* Java same way https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/main/java/org/sonar/java/checks/MethodTooBigCheck.java

**Comment by Bertk** [comment]
I will check this next week end. 
Replacing the line:
```
int lineCount = ChecksHelper.getRecursiveMeasureInt(sourceFunction, CxxMetric.LINES_OF_CODE); 
```
with the java plug-in approach might be possible.

**Comment by Bertk** [comment]
solved with #807.
